src/Modules/UngradedModules.jl

```diff
@@ -9089,3 +9089,48 @@ function has_monomials_on_all_axes(M::SubquoModule)
   return true
 end
 
+##################regularity#######################
+
+@doc raw"""
+    regularity(M::ModuleFP)
+
+Given a finitely presented graded module `M` over a standard $\mathbb Z$-graded 
+multivariate polynomial ring, return the Castelnuovo-Mumford regularity of `M`.
+ 
+# Examples
+```jldoctest
+julia> R, (x, y, z) = graded_polynomial_ring(QQ, ["x", "y", "z"]);
+
+julia> F = graded_free_module(R, 1);
+
+julia> M, _ = quo(F, [x^2*F[1], y^2*F[1], z^2*F[1]])
+(Graded subquotient of submodule of F generated by
+1 -> e[1]
+by submodule of F generated by
+1 -> x^2*e[1]
+2 -> y^2*e[1]
+3 -> z^2*e[1], F -> M
+e[1] -> e[1]
+Homogeneous module homomorphism)
+
+julia> regularity(M)
+3
+
+julia> minimal_betti_table(M);
+```
+"""
+function regularity(M::ModuleFP)
+ error("Not implemented for the given type")
+end
+
+function regularity(M::FreeMod)
+   return 0
+end
+
+function regularity(M::SubquoModule)
+   @assert is_standard_graded( base_ring(M))
```

Comment:
```suggestion
   @req is_standard_graded(base_ring(M)) "The base ring is not graded"
```
Adding a custom error message will be helpful for users.